src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Folding/FoldingRangeHandler.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RequestType } from 'vscode-languageclient';
+import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
+import { SerializableFoldingRangeParams } from '../RPC/SerializableFoldingRangeParams';
+import { SerializableFoldingRangeResponse } from '../RPC/SerializableFoldingRangeResponse';
+
+export class FoldingRangeHandler {
+    private static readonly provideFoldingRange = 'razor/foldingRange';
+    private foldingRangeRequestType: RequestType<SerializableFoldingRangeParams, SerializableFoldingRangeResponse, any> = new RequestType(FoldingRangeHandler.provideFoldingRange);
+
+    constructor(private readonly serverClient: RazorLanguageServerClient) {
+    }
+
+    public register() {
+        // tslint:disable-next-line: no-floating-promises
+        this.serverClient.onRequestWithParams<SerializableFoldingRangeParams, SerializableFoldingRangeResponse, any>(
+            this.foldingRangeRequestType,
+            async (request: SerializableFoldingRangeParams, token: vscode.CancellationToken) => this.provideFoldingRanges(request, token));
+    }
+
+    private async provideFoldingRanges(
+        foldingRangeParams: SerializableFoldingRangeParams,
+        cancellationToken: vscode.CancellationToken) {
+        // This is currently a no-op because we don't have a way to get folding ranges from C#/HTML.
```

Comment:
We should still be able to get Razor folding ranges, but C#/HTML ranges won't work in VSCode at this time :(

Reply:
Should we log an item for the backlog? Also, is there a quick explanation as to why? Or is it just that we can't yet call the Html or C# LSPs at all?

Reply:
VSCode needs to implement this work item - https://github.com/microsoft/vscode/issues/36638
Seems like it just hasn't been done yet (although looks like the work is scheduled now at least, which is good).

Good call on creating a work item on the Razor side, I'll do that and link it in the code.